libfreerdp/core/nla.c

```diff
@@ -200,22 +200,22 @@ static BOOL nla_adjust_settings_from_smartcard(rdpNla* nla)
 	if (!settings->SmartcardLogon)
 		return TRUE;
 
-	if (!smartcard_enumerateCerts(settings, &certs, &count))
+	if (!smartcard_enumerateCerts(settings, &nla->smartcardCerts, &nla->nsmartcardCerts))
```

Comment:
Might be a good idea to clean up with `smartcardCerts_Free` before starting the enumeration?